### PR TITLE
Adding Basic Chat Support

### PR DESCRIPTION
**Pull request by KevinSchildhorn**
## Details

Adding [Stream Chat](https://getstream.io/chat/docs/) support for Android and iOS. 

## Changes
* Adding `Stream` library for Android and iOS
* Adding new tab for Chat (This could be changed in the future)
* Adding Compose callback for calling the stream ui from shared ui
* Adding documentation
* Automatically signing the user into the `messaging` general` channel

| | Android | iOS |
| -- | ----- | ---- |
| compose | [Screen_recording_20240404_113702.webm](https://github.com/touchlab/DroidconKotlin/assets/5491479/fe742b2d-9986-4533-9a43-27453ffb0ed0) | Simulator Screen Recording - iPhone 15 Pro - 2024-04-04 at 10 21 42 |
| Swift UI| N/A | Simulator Screen Recording - iPhone 15 Pro - 2024-04-04 at 10 23 56 |

### CODE REVIEW

File: android/src/main/java/co/touchlab/droidcon/android/MainActivity.kt [hunk 1]
```diff
@@ -71,6 +74,17 @@ class MainActivity : ComponentActivity(), KoinComponent {
...
+                ChatManager.initChat(
+                    applicationContext = applicationContext,
+                    userData = UserData(
+                        id = user.uid,
+                        name = user.displayName,
+                        email = user.email,
+                        pictureUrl = user.photoUrl?.toString(),
+                    ),
+                )
+            }
```
Comment (faogustavo) [inline android/src/main/java/co/touchlab/droidcon/android/MainActivity.kt:87]: Is the chat only enabled for authenticated people? If yes, should we create a placeholder screen for the chat tab?

Reason: if I'm not authenticated, the app crashes

File: android/src/main/java/co/touchlab/droidcon/android/chat/ChatView.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                    )
+                }
+
+                InitializationState.INITIALIZING -> {
+                    Text(text = "Initialising...")
+                }
+
+                InitializationState.NOT_INITIALIZED -> {
+                    Text(text = "Not initialized...")
+                }
```
Comment (faogustavo) [inline android/src/main/java/co/touchlab/droidcon/android/chat/ChatView.kt:53]: Should we add an "action" to initialize or retry on failure?

File: android/src/main/java/co/touchlab/droidcon/android/MainActivity.kt (hunk 1 above)
Comment (faogustavo) [reply android/src/main/java/co/touchlab/droidcon/android/MainActivity.kt:87]: This logic does not update the current user when switching accounts.

**Comment by samhill303** [comment]
Closing PR's as we aren't planning to implement this right away. But will save the branch

**Pull request closed without merging**